### PR TITLE
page routes done

### PR DESCRIPTION
**Pull request by breakfireaus**

### CODE REVIEW

File: controllers/page-routes.js [hunk 1]
```diff
@@ -1,22 +1,162 @@
...
+  } catch (err) {
+    res.render('error', {
+      status: 500,
+      message: 'A internal server error has occured',
+      logged_in: req.session.logged_in,
+    });
+  }
+});
+
+router.get('/service/:id?mode=edit|create', withAuth, async (req, res) => {
```
Comment (Auralise) [inline controllers/page-routes.js]:
```suggestion
router.get('/edit-service', withAuth, async (req, res) => {

```
This route was specified differently in the document. Also, the query string parameter (? and everything after) is not included in the route definition, it is parsed in the function body by the res.query object. 

On reflection we need to change this route anyway as we are not going to know the ID of the service if we are creating a new one and thus was wrong when we specified it originally.  This page will have requests that look like the following: 

/edit-service?mode=create
/edit-service?mode=edit&id=5

We need to check if req.query contains mode or id. In the event of mode=create, render the edit-service page with just the logged_in status. If it contains mode=edit and no id value, call and internal server error as we don't know what we are editing and if it contains mode=edit and has an ID, look up the ID from the database and add it to the render call of the edit-service page.

We also need to check if the user attempting to edit a service is the same one who posted the service or we need to throw an error saying that they are unauthorised to edit the service. 

For reference of working with query strings: https://expressjs.com/en/api.html#req.query

**Review by Auralise — CHANGES_REQUESTED** [review]
On the whole, looking pretty good. Just one major change which was a bit of an issue with the initial specification, we assumed that we were going to know the ID of a service before we had created one. I have added some example calls to the endpoint as well as some guidance on how to get this working. 

Perhaps, as we need this file for working on the other components of the front end, can you please remove the route for edit service, commit and push again, and open a new branch for the edit-service route and put it in please? 

Nice work otherwise

**Comment by breakfireaus** [comment]
accepted change

**Comment by breakfireaus** [comment]
which lines you want me to remove
xxx - xxx

x = line number @Auralise

**Review by Auralise — APPROVED** [review]
We will merge this as is now and modify when we do some debugging and refactoring. 

All the rest should be good for the moment.

**Pull request merged**